connector/keystone/keystone_test.go

```diff
@@ -0,0 +1,326 @@
+package keystone
+
+import (
+	"testing"
+	"github.com/dexidp/dex/connector"
+
+	"fmt"
+	"io"
+	"os"
+  	"time"
+  	"net/http"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+   	networktypes "github.com/docker/docker/api/types/network"
+  	"github.com/docker/go-connections/nat"
+	"golang.org/x/net/context"
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"github.com/stretchr/testify/assert"
+)
+
+const dockerCliVersion = "1.37"
+
+const adminUser = "demo"
+const adminPass = "DEMO_PASS"
+const invalidPass = "WRONG_PASS"
+
+const testUser = "test_user"
+const testPass = "test_pass"
+const testEmail = "test@example.com"
+const testGroup = "test_group"
+
+const domain = "default"
+
+func startKeystoneContainer() string {
```

Comment:
This is a nice integration test suite.

But let's thread in a `t *testing.T` argument use `t.Helper()` and `t.Fatal` in the helper methods, instead of `fmt.Printf` calls, please 😉